File: app/src/main/java/com/devhyesun/kolinsample/api/model/GithubRepo.java
package com.devhyesun.kolinsample.api.model;

import com.google.gson.annotations.SerializedName;

public final class GithubRepo {

    public final String name;

    @SerializedName("full_name")
    public final String fullName;

    public final GithubOwner owner;

    public final String description;

    public final String language;

    @SerializedName("updated_at")
    public final String updateAt;

    @SerializedName("starazers_count")
    public int stars;

    public GithubRepo(String name, String fullName, GithubOwner owner, String description, String language, String updateAt, int stars) {
        this.name = name;
        this.fullName = fullName;
        this.owner = owner;
        this.description = description;
        this.language = language;
        this.updateAt = updateAt;
        this.stars = stars;
    }
}

File: app/src/main/kotlin/com/devhyesun/kolinsample/api/model/GithubRepo.kt
package com.devhyesun.kolinsample.api.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "repositories")
class GithubRepo(
    val name: String,
    @field:SerializedName("full_name")
    @PrimaryKey @ColumnInfo(name = "full_name") val fullName: String,
    @Embedded val owner: GithubOwner,
    val description: String?,
    val language: String?,
    @field:SerializedName("updated_at") val updateAt: String,
    @field:SerializedName("stargazers_count") var stars: Int)
